src/components/Reports/ReportsPageRanking/ReportsPageRanking.js

```diff
@@ -1,108 +1,110 @@
-      domainName: domainName,
-      dateFrom: dateFrom,
-    });
-    const pages = await this.asyncRequest;
-    this.asyncRequest = null;
-    this.setState({
-      pages: pages.value,
-    });
-  }
+  return (
+    <div className="wrapper-reports-box">
+      {state.loading ? (
+        <Loading />
+      ) : (
+        <S.ReportBox>
+          <small className="title-reports-box">
+            <FormattedMessage id="reports_pageranking.top_pages" />
+          </small>
 
-  componentWillUnmount() {
-    // TODO: abort request or at least side effects after finish
-    this.asyncRequest = null;
-  }
-
-  componentDidMount() {
-    this.fetchPagesRankingByPeriod(this.props.domainName, this.props.dateFrom);
-  }
-
-  static getDerivedStateFromProps(props, state) {
-    if (
-      props.domainName !== state.prevDomainName ||
-      props.dateFrom !== state.prevDateFrom ||
-      props.dateTo !== state.prevDateTo
-    ) {
-      return {
-        pages: null,
-        prevDomainName: props.domainName,
-        prevDateFrom: props.dateFrom,
-        prevDateTo: props.dateTo,
-      };
-    }
-
-    // No state update necessary
-    return null;
-  }
-
-  componentDidUpdate(prevProps, prevState) {
-    if (this.state.pages === null) {
-      this.fetchPagesRankingByPeriod(this.props.domainName, this.props.dateFrom);
-    }
-  }
-
-  render() {
-    const {
-      state: { pages },
-    } = this;
-
-    return (
-      <>
-        <div className="wrapper-reports-box">
-          {pages === null ? (
-            <Loading />
-          ) : (
-            <div className="reports-box">
-              <small className="title-reports-box">
-                <FormattedMessage id="reports_pageranking.top_pages" />
-              </small>
-
-              {pages.map((item, index) => (
-                <div key={index} className="page-ranking--item">
-                  <p className="text-ranking">
+          {state.pages ? (
+            state.pages.map((item, index) => (
+              <S.ListItem key={index}>
+                <S.ListItemColumn>
+                  <p>
                     <strong>{index + 1}</strong>
                   </p>
-                  <a
-                    className="link-ranking"
-                    href={item.url}
-                    target="_blank"
-                    rel="noopener noreferrer"
-                  >
+                  <a href={item.url} target="_blank" rel="noopener noreferrer">
                     {item.name}
                   </a>
-                  <p className="text-ranking">
+                  <p>
                     <strong>{item.totalVisitors}</strong>{' '}
                     <FormattedMessage id="reports_pageranking.total_visits" />
                   </p>
-                </div>
-              ))}
+                </S.ListItemColumn>
+                {experimentalFeatures &&
+                experimentalFeatures.getFeature('ReportsPageRankingVisits') ? (
```

Comment:
Maybe you do not need the experimental feature, what do you think about showing this column only if DataHub returns the expected values?

In that case, we could release the full version, and then, when DataHub starts returning the values, it will start working without our intervention.

What do you think?

Reply:
yes, i changed now and only validate the data